app/src/main/java/appricottsoftware/clarity/fragments/SettingFragment.java

```diff
@@ -52,11 +58,68 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         // Initialize view lookups, listener
         // TODO: write a listener for when fragment is done drawing UI
         tvProgress.setVisibility(View.GONE);
+        setUp();
+    }
+
+    private void setUp() {
         getPodcastLength();
         setSeekBar();
         getOldEmail();
         setEmailListener();
         setPasswordListener();
+        setDeleteAccountListener();
+    }
+
+    private void setDeleteAccountListener() {
+        final int uid = ClarityApp.getSession(getContext()).getUserID();
+        btDeleteAccount.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Log.i(TAG, "User clicked on Delete Account Button");
+                // Creating an
+                AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+                builder.setCancelable(true);
+                builder.setTitle("Delete Your Account?");
+                builder.setMessage("\nRemoving this account will PERMANENTLY DELETE all user's data from Clarity! \n\n" +
+                                    "Are you sure you want to PERMANENTLY delete your account?");
```

Comment:
Can you condense this message? Something like "Are you sure you want to permanently delete your account?".

Reply:
Done